src/multipart.rs

```diff
@@ -314,10 +314,38 @@ impl Stream for Multipart {
                 return Poll::Ready(None);
             }
 
-            if &boundary_bytes[..] != format!("{}{}{}", constants::BOUNDARY_EXT, boundary, constants::CRLF).as_bytes() {
-                return Poll::Ready(Some(Err(crate::Error::IncompleteStream)));
+            if &boundary_bytes[..] == format!("{}{}{}", constants::BOUNDARY_EXT, boundary, constants::CRLF).as_bytes() {
+                state.stage = StreamingStage::ReadingFieldHeaders;
             } else {
+                state.stage = StreamingStage::ReadingTransportPadding;
```

Comment:
This... doesn't seem quite right. I believe this will accept up to two additional arbitrary bytes:

```
--boundaryzz   \t\r\n
<data>
```

However, RFC 2046 does also say:

```
NOTE TO IMPLEMENTORS:  Boundary string comparisons must compare the
   boundary value with the beginning of each candidate line.  An exact
   match of the entire candidate line is not required; it is sufficient
   that the boundary appear in its entirety following the CRLF.
```

This "NOTE" has confused me more than if it had been left out. The boundary must take the form `--boundary[optional LWSP]CRLF`, but it is also sufficient to check only for `--boundary[...]`?

Reply:
The reason this is tricky for parsing is because we need to grab the two bytes following the boundary in case they are `--` (ending the stream). However if they are anything other than `--` those two bytes need to be "un-consumed" before transitioning to `ReadingTransportPadding` or handled directly here.